lib/operations/collection_ops.js

```diff
@@ -18,13 +18,12 @@ const handleCallback = require('../utils').handleCallback;
 const indexInformationDb = require('./db_ops').indexInformation;
 const isObject = require('../utils').isObject;
 const Long = require('mongodb-core').BSON.Long;
-const makeLazyLoader = require('../utils').makeLazyLoader;
 const MongoError = require('mongodb-core').MongoError;
 const ReadPreference = require('mongodb-core').ReadPreference;
 const toError = require('../utils').toError;
 
-const loadCollection = makeLazyLoader(`${__dirname}/../collection`);
-const loadDb = makeLazyLoader(`${__dirname}/../db`);
+const loadCollection = () => require('../collection');
```

Comment:
This will result in the same problem we were trying to fix, where the lazy loading will introduce a performance hit.

What we need to do is memoize the lazy loading function so that it only calls `require` once:

```js
let collection;
function loadCollection() {
  if (!collection) {
    collection = require('../collection');
  }
  return collection;
}
```